bls/src/main/java/tech/pegasys/artemis/util/bls/BLSSecretKey.java

```diff
@@ -15,12 +15,14 @@
 
 import java.util.Objects;
 import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes48;
 import tech.pegasys.artemis.util.mikuli.SecretKey;
 
 public final class BLSSecretKey {
 
   public static BLSSecretKey fromBytes(Bytes bytes) {
-    return new BLSSecretKey(SecretKey.fromBytes(bytes));
+    final Bytes keyBytes = bytes.size() == 32 ? Bytes48.leftPad(bytes) : bytes;
```

Comment:
do we want to raise an illegalargexception if size is < 32?

Reply:
Probably to do that here actually. I guess was thinking mikuli will do that check, but not nice relying on mikuli to verify that.

Reply:
mikuli only does check for size == 48

Reply:
so it should be ok to leave it as it is.

Reply:
think it would have been safe as it was, as mikuli will throw. but I like it checking the size in this class.
have added a size check

Reply:
👍